src/Analyser/IgnoreErrorExtension.php

```diff
@@ -0,0 +1,32 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Analyser;
+
+use PhpParser\Node;
+
+/**
+ * This is the extension interface to implement if you want to ignore errors
+ * based on the node and scope.
+ *
+ * To register it in the configuration file use the `phpstan.ignoreErrorExtension` service tag:
+ *
+ * ```
+ * services:
+ * 	-
+ *		class: App\PHPStan\MyExtension
+ *		tags:
+ *			- phpstan.ignoreErrorExtension
+ * ```
+ *
+ * Learn more: https://phpstan.org
+ *
+ * @api
+ */
+interface IgnoreErrorExtension
+{
+
+	public const EXTENSION_TAG = 'phpstan.ignoreErrorExtension';
+
+	public function ignore(Error $error, ?Node $node, Scope $scope): bool;
```

Comment:
Methods with names like `ignore` should return `void`. Method that returns `bool` should be named like `shouldIgnore` or `isIgnored(¨.